### PR TITLE
Progress without jQuery

### PR DESCRIPTION
**Pull request by levithomason**
Addresses the checklist item in #247. This PR updates Progress to our v1 API and removes jQuery.

See the doc examples for the API and usage.  I'd really appreciate reviewers pulling this branch and testing out the examples at [localhost:8080/#Progress](http://localhost:8080/#Progress).

### CODE REVIEW

File: docs/app/index.ejs [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <meta charset="UTF-8">
   <!-- baseHref value already includes quotes "/foo" -->
   <base href=<%= htmlWebpackPlugin.options.baseHref %> />
-  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/<%= htmlWebpackPlugin.options.versions.highlightjs %>/styles/github.min.css">
+  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.5.0/styles/github-gist.min.css">
```
Comment (levithomason) [inline docs/app/index.ejs:7]: We use react-highlight for syntax highlighting.  We'll replace it some day with a live editor.  For now, we manually update the highlight CSS version here.

File: src/modules/Progress/Progress.js [hunk 2]
```diff
@@ -1,110 +1,180 @@
-import $ from 'jquery'
-import classNames from 'classnames'
-import React, { Component, PropTypes } from 'react'
+import _ from 'lodash'
```
Comment (levithomason) [inline src/modules/Progress/Progress.js:1]: This file is a complete rewrite and best [viewed directly](https://github.com/TechnologyAdvice/stardust/blob/feature/progress-without-jquery/src/modules/Progress/Progress.js).

File: src/utils/jquery.js [hunk 3]
```diff
@@ -26,6 +26,3 @@ require('semantic-ui-css/components/checkbox')
 
 debug('Loading SUI Form plugin')
 require('semantic-ui-css/components/form')
-
-debug('Loading SUI Progress plugin')
-require('semantic-ui-css/components/progress')
```
Comment (levithomason) [inline src/utils/jquery.js:31]: One more down!  Only 2 left 🎉

File: docs/app/Components/ComponentDoc/ComponentExample.js [hunk 4]
```diff
@@ -50,15 +50,19 @@ export default class ComponentExample extends Component {
       marginLeft: '0.25em',
     }
 
+    const codeIconStyle = {
+      fontSize: '1.5em',
+      fontWeight: 'bold',
+    }
```
Comment (levithomason) [inline docs/app/Components/ComponentDoc/ComponentExample.js:56]: Cleaner code icon more inline with SUI docs.  Also removed the button.

image

File: docs/app/Examples/modules/Accordion/Types/Accordion.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React, { Component } from 'react'
 import { Accordion, Icon } from 'stardust'
 
-export default class AccordionStyledExample extends Component {
+export default class AccordionAccordionExample extends Component {
```
Comment (athurman) [inline docs/app/Examples/modules/Accordion/Types/Accordion.js:4]: typo? `AccordionAccordion`

Comment (levithomason) [inline docs/app/Examples/modules/Accordion/Types/Accordion.js:4]: No, just one of our ugly docs patterns.

``` js
${componentName}${exampleName}Example.js
```

So you get `AccrdionStyledExample.js` for instance.  But the first example name in SUI turns out to just be the component name, so we have `ButtonButtonExample.js` etc.  Docs are set to be overhauled and cleaned up once components are done.

**Pull request merged**